docs/content/v2025.1/architecture/transactions/concurrency-control.md

```diff
@@ -1271,3 +1271,73 @@ Refer to [#5680](https://github.com/yugabyte/yugabyte-db/issues/5680) for limita
 The `NOWAIT` clause for row-level explicit locking doesn't apply to the `Fail-on-Conflict` mode as there is no waiting. It does apply to the `Wait-on-Conflict` policy but is currently supported only for Read Committed isolation. [#12166](https://github.com/yugabyte/yugabyte-db/issues/12166) will extend support for this in the `Wait-on-Conflict` mode for the other isolation levels.
 
 The `SKIP LOCKED` clause is supported in both concurrency control policies and provides a transaction with the capability to skip locking without any error when a conflict is detected. However, it isn't supported for Serializable isolation. [#11761](https://github.com/yugabyte/yugabyte-db/issues/5683) tracks support for `SKIP LOCKED` in Serializable isolation.
+
+## YSQL lease mechanism for DML and DDL concurrency
+
+YugabyteDB employs a robust lease mechanism to ensure data consistency in a distributed environment, particularly during concurrent DML and DDL operations or YB-TServer failures. A YB-TServer must hold a valid lease from the YB-Master leader to serve any YSQL DMLs or DDLs.
```

Comment:
YugabyteDB employs a robust lease mechanism between TServer and Master, so that in the event a TServer is network partitioned from the master leader for a long time, the TServer is transitioned to a mode that doesn't allow the TServer to serve read or write traffic till it establishes connectivity with the master leader.